include/msgpack/v1/detail/cpp11_zone.hpp

```diff
@@ -25,153 +25,108 @@ namespace msgpack {
 /// @cond
 MSGPACK_API_VERSION_NAMESPACE(v1) {
 /// @endcond
-
+    
 class zone {
 private:
     struct finalizer {
-        finalizer(void (*func)(void*), void* data):m_func(func), m_data(data) {}
+        finalizer(void (*func)(void*), void* data, finalizer* next): m_func(func), m_data(data), m_next(next) {}
         void operator()() { m_func(m_data); }
         void (*m_func)(void*);
         void* m_data;
+        finalizer* m_next{};
     };
     struct finalizer_array {
-        finalizer_array():m_tail(MSGPACK_NULLPTR), m_end(MSGPACK_NULLPTR), m_array(MSGPACK_NULLPTR) {}
-        void call() {
-            finalizer* fin = m_tail;
-            for(; fin != m_array; --fin) (*(fin-1))();
+        finalizer_array(): m_head(MSGPACK_NULLPTR) {}
+
+        void call(bool clear = false) {
```

Comment:
```suggestion
        void call() {
```

There is only one caller `call(true)` so no parameter is required.